apps/studio/components/grid/components/header/Header.tsx

```diff
@@ -327,6 +327,52 @@ const RowHeader = ({ table, sorts, filters }: RowHeaderProps) => {
     setIsExporting(false)
   }
 
+  async function onRowsExportSQL() {
+    setIsExporting(true)
+
+    if (allRowsSelected && totalRows > MAX_EXPORT_ROW_COUNT) {
+      toast.error(
+        `Sorry! We're unable to support exporting of SQL for row counts larger than ${MAX_EXPORT_ROW_COUNT.toLocaleString()} at the moment.`
+      )
+      return setIsExporting(false)
+    }
+
+    if (!project) {
+      toast.error('Project is required')
+      return setIsExporting(false)
+    }
+
+    const rows = allRowsSelected
+      ? await fetchAllTableRows({
+          projectRef: project.ref,
+          connectionString: project.connectionString,
+          table,
+          filters,
+          sorts,
+          impersonatedRole: roleImpersonationState.role,
+        })
+      : allRows.filter((x) => selectedRows.has(x.idx))
+
+    const sqlStatements = rows
+      .map((row) => {
+        const filteredRow = { ...row }
+        delete filteredRow.idx
+
+        const columns = Object.keys(filteredRow)
+          .map((col) => `"${col}"`)
+          .join(', ')
+        const values = Object.values(filteredRow)
+          .map((val) => (typeof val === 'object' && val !== null ? JSON.stringify(val) : val))
+          .map((val) => `'${val}'`)
```

Comment:
hmm i'd be careful here RE correctness in representation of data
we'd technically stringify numerical values here too which, depending on the column's data type, might not be want the user wants

ideally we'd also have some unit tests around this to ensure its correctness since we're writing the exporting logic ourselves

Reply:
also this looks like `n` number of INSERT statements which i'm guessing might be rough on the DB esp if theres gonna be a lot of rows - we should export this into a single INSERT statement with multiple values imo

Reply:
i'm currently taking a look at this - will push up some changes